packages/davinci-client/src/lib/davinci.utils.ts

```diff
@@ -172,12 +172,25 @@ export function handleResponse(cacheEntry: DaVinciCacheEntry, dispatch: Dispatch
    */
   if (cacheEntry.isSuccess) {
     const requestId = cacheEntry.requestId;
-    if ('eventName' in cacheEntry.data && cacheEntry.data.eventName === 'continue') {
-      const data = cacheEntry.data as DaVinciNextResponse;
-      dispatch(nodeSlice.actions.next({ data, requestId, httpStatus: status }));
-    } else if ('session' in cacheEntry.data || 'authorizeResponse' in cacheEntry.data) {
+    const hasNextUrl = () => {
+      const data = cacheEntry.data;
+
+      if ('_links' in data) {
+        if ('next' in data._links) {
+          if ('href' in data._links.next) {
+            return true;
+          }
+        }
+      }
+      return false;
+    };
```

Comment:
I feel like this should be a pure, stateless utility that could be used for fetching any property within `_links`. This could be reused in other portions of the project.